esp-hal/src/i2s/master.rs

```diff
@@ -297,12 +300,15 @@ where
                 i2s: unsafe { i2s.clone_unchecked() },
                 rx_channel: channel.rx,
                 descriptors: rx_descriptors,
+                guard,
             },
             i2s_tx: TxCreator {
                 i2s,
                 tx_channel: channel.tx,
                 descriptors: tx_descriptors,
+                guard: PeripheralGuard::new(peripheral),
             },
+            guard: PeripheralGuard::new(peripheral),
```

Comment:
It's a small thing but I don't think the outer driver needs a guard. It costs a bit of memory and maybe a few cycles, and the Tx/Rx guard are sufficient.